src/version/VersionContext.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useContext, useEffect, useState } from "react";
+import axios from 'axios';
+import { Children } from "../common/types/Helpers";
+import { VERSION } from "./version";
+
+export interface Version {
+    version: number;
+    releaseNotes: string;
+}
+
+export interface VersionContext {
+    currentVersion: number;
+    latestVersion?: Version;
+}
+
+const VersionContextObject: React.Context<VersionContext> = React.createContext(initialContextValue());
+
+function initialContextValue(): VersionContext {
+    return {
+        currentVersion: VERSION
+    };
+}
+
+export interface Props {
+    children?: Children;
+}
+
+export function VersionContextProvider(props: Props) {
+    const [ contextValue, setContextValue ] = useState<VersionContext>(initialContextValue());
+    const [ fetchedVersion, setFetchedVersion ] = useState(false);
+
+    useEffect(() => {
+        if(!fetchedVersion) {
+            setFetchedVersion(true);
+            (async function() {
+                const latestVersion = await fetchVersion();
+                setContextValue({
+                    ...contextValue,
+                    latestVersion
+                });
+            })();
+        }
+    }, [ fetchedVersion, setFetchedVersion, contextValue, setContextValue ]);
+
+    return (
+        <VersionContextObject.Provider value={contextValue}>
+            {props.children}
+        </VersionContextObject.Provider>
+    );
+}
+
+async function fetchVersion(): Promise<Version> {
+    try {
+        const response = await axios.get(`${process.env.PUBLIC_URL}/version.json?timestamp=${new Date().getTime()}`);
```

Comment:
I guess the parameter`timestamp` is used to avoid caching. Not sure it's necessary, a similar mechanism is used to refresh LOC when waiting for the timestamps, and it does work very nicely (`304` http code returned as long as we don't have `addedOn`, `200` as soon as it's in the payload).

Reply:
Yes it is. Maybe, but most people trying to prevent client-side caching are using that approach (e.g. see [here](https://stackoverflow.com/a/49263912)). I'll keep it like that, just to be sure...